### ISSUE TITLE
cannot reproduce examples/pytorch/HAN 

### ISSUE THREAD

**Issue by lipingcoding** [dmlc/dgl#2385]
## To Reproduce

1. python main.py
1.
1.

Test loss 0.4159 | Test Micro f1 0.8748 | Test Macro f1 0.8741

## Expected behavior

Test Micro f1 0.8899 | Test Macro f1 0.8902

## Environment

 - DGL Version (e.g., 0.5.3): 0.5.3
 - Backend Library & Version (e.g., PyTorch 1.7.0, ): pytorch 1.7.0
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source): pip 
 - Build command you used (if compiling from source):
 - Python version: 3.7.0
 - CUDA/cuDNN version (if applicable): cuda 10.1
 - GPU models and configuration (e.g. V100): Titan XP
 - Any other relevant information:

**Comment by mufeili** [comment]
Can you try running multiple times? There can be variance across runs due to randomly initialized weights.

**Comment by lipingcoding** [comment]
I have run multiple times, but every time I got the same result. What's more, I have tried dgl 0.4.3 and corresponding han example code, I got better result, but still worse than dgl and paper reported

**Comment by lipingcoding** [comment]
Just now, I have tried different seeds for main.py. I found the choice of seed has an unbelievable impact on performance. 


seed 2    Test loss 0.3230 | Test Micro f1 0.8814 | Test Macro f1 0.8816

seed 4 Test loss 0.3350 | Test Micro f1 0.8758 | Test Macro f1 0.8759

seed 10 Test loss 0.4631 | Test Micro f1 0.8494 | Test Macro f1 0.8484

I am not sure whether it is the property of the HAN model itself, or there are some bug in dgl HAN example code.

**Comment by yzh119** [comment]
The model's performance would easily be influenced by random seed for most GNN models on small datasets like ACM.

**Comment by yzh119** [comment]
The number in README.md was obtained 11 months ago, the corresponding dgl version is 0.4.2, you can checkout the example at that time: https://github.com/dmlc/dgl/tree/3d6646288fcbbeb3e06e840ad26d5a8f1b665081/examples/pytorch/han to exactly reproduce the number.

The codebase of DGL has been refactored for several time since then and we cannot make sure each model could yield the same performance as before (especially on small datasets where hyper-parameters such as random seed really matters). I also tried `python main.py --hetero` and got test macro f1 `0.9209` which is higher than `0.9166` reported in the table.

**Comment by yzh119** [comment]
@lipingcoding #1790 might be related, where the Semantic Attention class was changed to: 
![image](https://user-images.githubusercontent.com/11773619/100878396-176b5f80-34e5-11eb-9931-4bf875372c2c.png)

I changed it back and got a test macro f1 of `0.8889` by setting the random seed to 2, you can try other random seeds and I believe there is a seed that achieves a f1 greater than 0.89.
